app/[locale]/@modals/(.)search/page.tsx

```diff
@@ -1,10 +1,14 @@
 import Search from '~/app/search/search';
 import { Dialog } from '~/components/dialog';
 
-export default function Page() {
+export default function Page({
+  searchParams,
+}: {
+  searchParams: Record<string, string | undefined>;
```

Comment:
In fact, I would recommend this for code clarity:
```suggestion
  searchParams: { q: query },
}: {
  searchParams: { q: string };
```
This will abstract out `q` from the `searchParams` object, letting you use `q` directly in the code. And `q: query` will essentially rename `q` to `query`.